### PR TITLE
[2.9] Fix issue with MongoDB 4.0.20

### PR DESCRIPTION
**Pull request by rhysmeister**
##### SUMMARY
Update to module and tests to fix https://github.com/ansible-collections/community.mongodb/issues/136

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
mongodb_replicaset & tests for mongodb_replicaset and mongodb_shard

### CODE REVIEW

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 1]
```diff
@@ -352,65 +350,61 @@ def main():
...
-            try:
-                rs = replicaset_find(client)
-            except Exception as e:
-                module.fail_json(msg='Unable to query replica_set info: %s' % to_native(e))
-            if rs is not None and rs != replica_set:
-                module.fail_json(msg="The replica_set name of '{0}' does not match the expected: '{1}'".format(rs, replica_set))
-        result['changed'] = False
-
-    module.exit_json(**result)
+        module.fail_json(msg='Unable to connect to database: %s' % to_native(e))
```
Comment (nitzmahone) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: This is an awfully big exception handler with a really specific failure message- any unhandled exception through most of the module body will fail with "Unable to connect to database". The smaller sub-operation try/except blocks mitigate that somewhat, but it should probably just be limited to the connection bits as it was previously (so any future additions inside the try block won't fail with a misleading error).

**Review by nitzmahone — CHANGES_REQUESTED** [review]
One minor structural question, and needs a changelog entry, but otherwise this should be fine.

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py (hunk 1 above)
Comment (rhysmeister) [reply lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: Noted and modified.

**Review by nitzmahone — APPROVED** [review]

**Comment by nitzmahone** [comment]
@rhysmeister looks great. I truncated the changelog text a bit and added the module name to the front (usually just need the affected component, a terse description, and link to the issue). We'll get this merged for the next 2.9 update. Thanks!

**Pull request merged**